### PR TITLE
[CMAKE] Improve configure script to properly work with newer CMake ve…

### PR DESCRIPTION
**Pull request by tkreuzer**
…rsions

## Purpose

This change allows to properly pass parameters to newer CMake versions. Currently the script passes "VSSolution" and similar on to CMake, which are ignored, but newer CMake will complain. Works for me to configure a VSSolution with CMake 3.27.4. More testing appreciated.

### CODE REVIEW

File: configure.cmd [hunk 1]
```diff
@@ -75,6 +75,21 @@ if not defined ARCH (
...
 set USE_CLANG_CL=0
 
+REM Some magic code to parse the parameters and extract those for CMake
+set CMAKE_PARAMS=
+set REMAINING=%*
+:loop
+for /f "tokens=1*" %%a in ("%REMAINING%") do (
+    set "PARAM=%%a"
+    if "!PARAM:~0,1!" == "-" (
+        if "!PARAM!" NEQ "-VS_VER" (
```
Comment (tkreuzer) [inline configure.cmd:85]: Alternatively we could change the parameter to be `VS_VER=xx` without the dash to make it more consistent.

File: configure.cmd [hunk 2]
```diff
@@ -118,6 +133,8 @@ REM Parse command line parameters
             REM explicitly set VS version for project generator
             if /I "%2" == "-VS_VER" (
                 set VS_VERSION=%3
+                SHIFT
+                SHIFT
```
Comment (binarymaster) [inline configure.cmd:137]: Nice command, didn't knew about it. Maybe `SHIFT /n 2` instead of two commands?

File: configure.cmd [hunk 3]
```diff
@@ -75,6 +75,21 @@ if not defined ARCH (
 
 set USE_CLANG_CL=0
 
+REM Some magic code to parse the parameters and extract those for CMake
+set CMAKE_PARAMS=
+set REMAINING=%*
+:loop
+for /f "tokens=1*" %%a in ("%REMAINING%") do (
```
Comment (learn-more) [inline configure.cmd:82]: Wouldnt it be better to merge it into command line parsing thing below?

File: configure.cmd [hunk 4]
```diff
@@ -198,7 +215,7 @@ if "%BUILD_ENVIRONMENT%" == "MinGW" (
 ) else if %USE_CLANG_CL% == 1 (
     cmake -G %CMAKE_GENERATOR% -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=%ARCH% %BUILD_TOOLS_FLAG% -DUSE_CLANG_CL:BOOL=1 %* "%REACTOS_SOURCE_DIR%"
```
Comment (learn-more) [inline configure.cmd:216]: The other cmake invocations probably should probably also be updated?

File: configure.cmd (hunk 3 above)
Comment (tkreuzer) [reply configure.cmd]: Done.

File: configure.cmd (hunk 4 above)
Comment (tkreuzer) [reply configure.cmd]: Fixed

File: configure.cmd (hunk 2 above)
Comment (tkreuzer) [reply configure.cmd]: Removed it entirely

**Review by learn-more — APPROVED** [review]
lgtm.
Without patch:
`cmake  -G "Visual Studio 17" -A Win32 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=i386  VSSolution -DENABLE_ROSTESTS=1 -DENABLE_ROSAPPS=1 "R:\src\dev\"`

With:
`cmake  -G "Visual Studio 17" -A Win32 -DCMAKE_TOOLCHAIN_FILE:FILEPATH=toolchain-msvc.cmake -DARCH:STRING=i386   -DENABLE_ROSTESTS=1 -DENABLE_ROSAPPS=1 "R:\src\dev\"`

**Pull request merged**